### PR TITLE
upgrade to 5.0.0-alpha4

### PR DESCRIPTION
**Pull request by imotov**
This commit also removes analyzed_text mapper, naive bayes update script and spark mllib dependencies, which turned out to be too problematic for the upgrade. Since in 5.0 only one scripting engine per plugin is allowed the prep spec scripting engine was transformed into a native script.

### CODE REVIEW

File: src/main/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchContext.java [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
- */
-
-package org.elasticsearch.search.fetch.analyzedtext;
-
-
-import org.elasticsearch.action.admin.indices.analyze.AnalyzeRequest;
-import org.elasticsearch.action.termvectors.TermVectorsRequest;
-import org.elasticsearch.search.fetch.FetchSubPhaseContext;
-
-public class AnalyzedTextFetchContext extends FetchSubPhaseContext {
```
Comment (brwe) [inline src/main/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchContext.java:27]: Wait. Why does this have to go too?

File: src/test/java/org/elasticsearch/action/preparespec/PrepareSpecIT.java [hunk 2]
```diff
@@ -52,61 +52,79 @@
...
         assertThat(entries.isSparse(), equalTo(false));
         assertThat(entries.getEntries().size(), equalTo(1));
     }
 
-    private void indexDocs() {
-        client().admin().indices().prepareCreate("index").setSettings(Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)).get();
+    private void indexDocs() throws IOException {
+        XContentBuilder mapping = jsonBuilder();
+        mapping.startObject();
+        {
```
Comment (brwe) [inline src/test/java/org/elasticsearch/action/preparespec/PrepareSpecIT.java:94]: nice trick :)

File: src/test/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchIT.java [hunk 3]
```diff
@@ -1,75 +0,0 @@
...
-import java.util.Collection;
-import java.util.List;
-
-import static org.elasticsearch.client.Requests.indexRequest;
-import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
-
-
-public class AnalyzedTextFetchIT extends ESIntegTestCase {
-
```
Comment (brwe) [inline src/test/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchIT.java:40]: see above. if we can keep the fetch phase we need to keep the test too.

File: src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java [hunk 4]
```diff
@@ -24,66 +24,44 @@
...
 
 import java.io.IOException;
 
 public class TrainNaiveBayesResponse extends ActionResponse implements ToXContent {
 
-    public String getIndex() {
-        return index;
-    }
-
-    public String getType() {
```
Comment (brwe) [inline src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java:37]: Should we really not return type and index? I found it convenient when I wanted to look at the script after it was trained. I always forget whether the index is .scripts or .script and also was never sure which was the name of the language.

File: src/main/java/org/elasticsearch/action/trainnaivebayes/TransportTrainNaiveBayesAction.java [hunk 5]
```diff
@@ -267,12 +270,25 @@ public Object run() {
...
+            }
+            client.admin().cluster().preparePutStoredScript().setScriptLang(PMMLModelScriptEngineService.NAME)
+                    .setSource(source).setId(id)
+                    .execute(new ActionListener<PutStoredScriptResponse>() {
                         @Override
-                        public void onResponse(IndexResponse indexResponse) {
-                            listener.onResponse(new TrainNaiveBayesResponse(indexResponse.getIndex(), indexResponse.getType(),
-                                    indexResponse.getId()));
+                        public void onResponse(PutStoredScriptResponse indexResponse) {
+                            //TODO: Fix response
```
Comment (brwe) [inline src/main/java/org/elasticsearch/action/trainnaivebayes/TransportTrainNaiveBayesAction.java]: what does this mean?

**Comment by brwe** [comment]
Again, you cannot imagine how happy this makes me! I do not think it makes sense to review much because the code was untidy before and it is just a prototype anyway. I left a few questions though.

File: src/main/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchContext.java (hunk 1 above)
Comment (imotov) [inline src/main/java/org/elasticsearch/search/fetch/analyzedtext/AnalyzedTextFetchContext.java:27]: Oops, got carried away, I guess.

File: src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java (hunk 4 above)
Comment (imotov) [inline src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java:37]: There is not index anymore elastic/elasticsearch/pull/17650. Scripts are stored in the cluster state now.

File: src/main/java/org/elasticsearch/action/trainnaivebayes/TransportTrainNaiveBayesAction.java (hunk 5 above)
Comment (imotov) [inline src/main/java/org/elasticsearch/action/trainnaivebayes/TransportTrainNaiveBayesAction.java]: That's means to remove index and type, which I did. But I forgot to remove TODO. Thanks for catching it.

**Comment by imotov** [comment]
I've restored the analyzed text sub phase that I accidentally deleted. Sorry about that.

File: src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java (hunk 4 above)
Comment (brwe) [inline src/main/java/org/elasticsearch/action/trainnaivebayes/TrainNaiveBayesResponse.java:37]: doh! time flies. I thought we just discussed it yesterday and totally missed it was merged already :)

**Comment by brwe** [comment]
LGTM!

**Pull request merged**